src/python/tests/core/bot/fuzzers/libFuzzer/engine_test.py

```diff
@@ -59,11 +71,176 @@ def test_prepare(self):
     engine_impl = engine.LibFuzzerEngine()
     options = engine_impl.prepare('/corpus_dir', '/path/target', '/path')
     self.assertEqual('/corpus_dir', options.corpus_dir)
-    self.assertItemsEqual(
-        ['-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1'],
-        options.arguments)
+    self.assertItemsEqual([
+        '-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1',
+        '-dict=/path/blah.dict'
+    ], options.arguments)
     self.assertItemsEqual(['strategy1', 'strategy2'], options.strategies)
-    self.assertItemsEqual(['/new_corpus_dir'], options.additional_corpus_dirs)
+    self.assertItemsEqual(['/new_corpus_dir', '/corpus_dir'],
+                          options.fuzz_corpus_dirs)
     self.assertDictEqual({'extra_env': '1'}, options.extra_env)
     self.assertFalse(options.use_dataflow_tracing)
     self.assertTrue(options.is_mutations_run)
+
+    self.mock.unpack_seed_corpus_if_needed.assert_called_with(
+        '/path/target', '/corpus_dir')
+
+  def test_prepare_invalid_dict(self):
+    """Test prepare with an invalid dict path."""
+    with open('/path/target.options', 'w') as f:
+      f.write('[libfuzzer]\n'
+              'max_len=31337\n'
+              'timeout=11\n'
+              'dict=nope.dict\n')
+
+    engine_impl = engine.LibFuzzerEngine()
+    options = engine_impl.prepare('/corpus_dir', '/path/target', '/path')
+    self.assertItemsEqual(
+        ['-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1'],
+        options.arguments)
+
+  def test_prepare_auto_add_dict(self):
+    """Test prepare automatically adding dict argument."""
+    with open('/path/target.options', 'w') as f:
+      f.write('[libfuzzer]\n' 'max_len=31337\n' 'timeout=11\n')
+    self.fs.create_file('/path/target.dict')
+
+    engine_impl = engine.LibFuzzerEngine()
+    options = engine_impl.prepare('/corpus_dir', '/path/target', '/path')
+    self.assertItemsEqual([
+        '-max_len=31337', '-timeout=11', '-rss_limit_mb=2048', '-arg1',
+        '-dict=/path/target.dict'
+    ], options.arguments)
+
+
+class FuzzTest(fake_fs_unittest.TestCase):
+  """Fuzz() tests."""
+
+  def setUp(self):
+    # Set up fake filesystem.
+    test_helpers.patch_environ(self)
+    test_utils.set_up_pyfakefs(self)
+
+    self.fs.create_dir('/corpus')
```

Comment:
we probably need to do a mass rename of self.fs.CreateDirectory and self.fs.CreateFile, probably can do this after this cl ? or i can do it.

Reply:
Let's do this in another CL.